Core/clim-core/panes.lisp

```diff
@@ -2690,11 +2675,12 @@ SCROLLER-PANE appear on the ergonomic left hand side, or leave set to
       ;;
       ;; The solution is to queue an event that will trigger the scoll
       ;; later, at which point the output records are properly setup.
-      (queue-event stream
-                   (make-instance 'scroll-extent-event
-                                  :sheet stream
-                                  :stream stream
-                                  :new-height new-height)))))
+      (scroll-extent stream
+                   0
+                   (max 0 (- new-height
+                             (bounding-rectangle-height
+                              (or (pane-viewport stream)
+                                  stream))))))))
```

Comment:
according to comment above this was queued to prevent race when drawing. how does making this operation immediate prevent the problem? also, if it does, then the comment above is not adequate anymore, so in such situation it should be removed too.

Reply:
I made that change (and wrote that comment) as part of that work. Rolling this back means that we're back where we started.

The issue that that particular code fixes is real. It ensures that any repainting that happens as part of the scroll will draw the correct content. However, without the optimised scroll, this never happens. 

Why remove it then? Well, the current implementation is incomplete when there are more than one scroll between repaints. If the second scroll operation looks at the position, it will see the old coordinate (since the actual scroll hasn't happened yet) and the resulting scroll position will be incorrect. It's rare, but it can happen in an interactor pane.

This can be solved by adding more complexity, but I figures since nothing needs this as the moment its better to roll back to what worked before I started working on the scroll optimisation.

Reply:
so this comment should be modified, because it describe a measurement which is no longer there

Reply:
Indeed. It should have been removed along with the code change. That was a mistake. Thanks for pointing it out. I have removed it as of a04e9c2ab708f53971b97cbecb3da2617ef05e34